config/initializers/user_action_events.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module UserActionEvents
```

Comment:
initializers typically aren't modularized however, I think it might be more appropriate to put all this logic inside it's own class and call it in `application.rb` like in a `middleware` or even a `Railtie`. That way we can ensure the app is spun up and the database is connected and we wont have to do all the extra checks. Like what you have at the end  here would make more sense in `application.rb`

```ruby
class Application < Rails::Application
  config.after_initialize do
    SomeClass.load
  end
end